test/sqllogictest/transactions.slt

```diff
@@ -187,3 +187,84 @@ simple
 COMMIT
 ----
 db error: ERROR: unknown catalog item 'u1'
+
+# Verify TAIL must be only read statement in a transaction.
+# We use FETCH 0 with TAIL below so that we don't need to worry about
+# timestamps in this slt file.
+
+statement ok
+CREATE TABLE t (a int)
+
+statement ok
+INSERT INTO t VALUES (1)
+
+simple
+DECLARE c CURSOR FOR TAIL t;
+FETCH 0 c;
+DECLARE d CURSOR FOR TAIL t;
+FETCH 0 d;
+----
+db error: ERROR: TAIL in transactions must be only read statement
+
+simple
+DECLARE c CURSOR FOR TAIL t;
+FETCH 0 c;
+SELECT * FROM t LIMIT 0;
+----
+db error: ERROR: TAIL in transactions must be only read statement
+
+simple
+SELECT * FROM t LIMIT 0;
+DECLARE c CURSOR FOR TAIL t;
+FETCH 0 c;
+----
+db error: ERROR: transaction in read-only mode
```

Comment:
This error would be confusing to the user as only reads are being done in this transaction.

Reply:
I agree. But I'm not sure what to change it to because we want to match postgres for this error type, and we've invented some new transaction modes that postgres doesn't have.